### DISCUSSION TITLE
useMagicKeys - meta/ctrl + k - firefox

### DISCUSSION THREAD

**Discussion by edtorba** [vueuse/vueuse#2636] [Q&A]
Hi all, got a little dilemma with the following setup. Am implementing a key combo of meta (cmd) (or ctrl) + k to toggle the modal window. The code below works but in Firefox it also focuses on the URL bar. 

**Question:** What would be the correct way to prevent that?

```js
import { useMagicKeys } from '@vueuse/core';
const keys = useMagicKeys();
const metaK = keys['meta+k'];

watch(metaK, (v) => {
  if (v) {
    // Do what I want.
  }
});
```

**Comment by c4mbr0nn3** [comment]
I also have this issue, both firefox and chrome. My snippet is the following:

```javascript
const { Ctrl_K } = useMagicKeys();

whenever(Ctrl_K, () => {
  dialog.value = true;
});
```

I didn't find a way to prevent default behaviour, is this expected and instead we should use `onKeyStroke`?

**Comment by s-montigny-desautels** [comment]
Does this work?

```ts
const keys = useMagicKeys({
    passive: false,
    onEventFired: (e) => {
        if (keys['meta+k']) {
            e.preventDefault();
        }
},
});
```

**Reply by simaotwx** [reply]
I tried Ctrl+Shift+N and it still opens a new browser window.

**Reply by s-montigny-desautels** [reply]
You can't prevent this keybind. It's not even passed to the javascript runtime in Chrome.

**Reply by simaotwx** [reply]
Oh damn. It did get passed through in Firefox, just that I couldn't prevent the browser from doing what it usually does.

**Reply by Bassadin** [reply]
[The docs explicitly warn against using this](https://vueuse.org/core/useMagicKeys/#custom-event-handler) - can it break anything?

**Reply by s-montigny-desautels** [reply]
I think it's only a warning about overriding system hotkey. See here: https://github.com/vueuse/vueuse/issues/429#issuecomment-816582038
